scripts/kind-build-test.sh

```diff
@@ -0,0 +1,154 @@
+            -v          Kubernetes Version (Default: 1.16) [1.14, 1.15, 1.16, 1.17, and 1.18]
+
+EOM
+)
+
+# Process our input arguments
+while getopts "ps:ioc:b:v:" opt; do
+  case ${opt} in
+    p ) # PRESERVE K8s Cluster
+        echo "❄️  This run will preserve the cluster as you requested"
+        PRESERVE=true
+      ;;
+    s ) # AWS Service name
+        echo "Running Docker build as you requested for ${OPTARG} service"
+        AWS_SERVICE=$(echo "${OPTARG}" | tr '[:upper:]' '[:lower:]')
+        echo $AWS_SERVICE
+      ;;
+    i ) # AWS Service Docker Image
+        AWS_SERVICE_DOCKER_IMG="${OPTARG}"
+      ;;
+    o ) # Override path with your own kubectl and kind binaries
+        DELETE_CLUSTER_ARGS="${DELETE_CLUSTER_ARGS} -o"
+        PROVISION_CLUSTER_ARGS="${PROVISION_CLUSTER_ARGS} -o"
+        OVERRIDE_PATH=1
+      ;;
+    c ) # Cluster context directory to operate on existing cluster
+        TMP_DIR="${OPTARG}"
+      ;;
+    b ) # Base cluster name
+        CLUSTER_NAME_BASE="${OPTARG}"
+      ;;
+    v ) # K8s VERSION
+        K8S_VERSION="${OPTARG}"
+      ;;
+    \? )
+        echo "${USAGE}" 1>&2
+        exit
+      ;;
+  esac
+done
+
+if [ -z $TMP_DIR ]; then
+    TMP_DIR=$("${SCRIPT_PATH}"/provision-kind-cluster.sh -b "${CLUSTER_NAME_BASE}" -v "${K8S_VERSION}" "${PROVISION_CLUSTER_ARGS}")
+fi
+
+if [ $OVERRIDE_PATH == 0 ]; then
+  export PATH=$TMP_DIR:$PATH
+else
+  export PATH=$PATH:$TMP_DIR
+fi
+
+CLUSTER_NAME=$(cat $TMP_DIR/clustername)
+
+## Build and Load Docker Images
+
+if [ -z "$AWS_SERVICE_DOCKER_IMG" ]; then
+    echo "🥑 Building ${AWS_SERVICE} docker image"
+    DEFAULT_AWS_SERVICE_DOCKER_IMG="${AWS_SERVICE}:${VERSION}"
+    docker build -f ${ROOT_DIR}/services/${AWS_SERVICE}/Dockerfile -t "${DEFAULT_AWS_SERVICE_DOCKER_IMG}" .
+    AWS_SERVICE_DOCKER_IMG="${DEFAULT_AWS_SERVICE_DOCKER_IMG}"
+    echo "👍 Built the ${AWS_SERVICE} docker image"
+else
+    echo "🥑 Skipping building the ${AWS_SERVICE} docker image, since one was specified ${AWS_SERVICE_DOCKER_IMG}"
+fi
+echo "$AWS_SERVICE_DOCKER_IMG" > "${TMP_DIR}"/"${AWS_SERVICE}"_docker-img
+
+echo "🥑 Loading the images into the cluster"
+kind load docker-image --name "${CLUSTER_NAME}" --nodes="${CLUSTER_NAME}"-worker,"${CLUSTER_NAME}"-control-plane "${AWS_SERVICE_DOCKER_IMG}"
+echo "👍 Loaded image(s) into the cluster"
+
+export KUBECONFIG="${TMP_DIR}/kubeconfig"
+
+trap "exit_and_fail" INT TERM ERR
+trap "clean_up" EXIT
+
+echo "======================================================================================================"
+echo "To poke around your test manually:"
+echo "export KUBECONFIG=$TMP_DIR/kubeconfig"
+echo "export PATH=$TMP_DIR:\$PATH"
+echo "kubectl get pods -A"
+echo "======================================================================================================"
+
+
+### exported vars and funcs that tests can use
+export TMP_DIR
+export CLUSTER_NAME
+export AEMM_URL
+export AEMM_VERSION
+export AEMM_DL_URL
```

Comment:
above three AEMM variables are unused (refer to the mock metadata service that we don't need)